src/LinAlg/ProcessAdm.h

```diff
@@ -33,7 +34,7 @@ class ProcessAdm
   int nProc;       //!< Number of processes in communicator
   bool parallel;   //!< If Processor is parallel
 
-#ifdef HAS_PETSC
+#if defined(HAS_PETSC) || defined(HAVE_MPI)
   MPI_Comm comm;   //!< MPI communicator
 #endif
```

Comment:
i DD-klassen er det "ren" mpi-kode, som er dekoblet fra petsc. dette kan brukes til feks å legge inn støtte for parallelle istl-løsere (jeg har ikke kommet i mål her ennå), eller andre lineærløsningspakker hvis det skulle være ønskelig.

MPI_Comm er et MPI-konsept som brukes av petsc. så derfor kan vi ha en MPI_Comm selv om petsc ikke fins. denne brukes i DD-opprettelsen (og i lineærløsningspakken).

den største grunnen til at ting blir litt kompleks er måten PETSc implementerer serielle løsere på. dette gjøres ved å ha et 'fake' mpi-bibliotek, mpiuni. dette emulerer mpi-kallet på single-prosess-systemer (essensielt memcpy) sånn at petsc-koden er alltid "parallell" i betydning bruker alltid MPI. i den forbindelse er det en del hacks, for mpiuni støtter ikke hele MPI. den gjør feks ting som
```C
 if (comm == PETSC_COMM_WORLD) 
  do_something;
else
  do_something_else;
```
følgelig må jeg bruke mpi sånn som petsc definerer det når petsc er aktiv, for det kan være mpiuni eller det kan være et reelt mpi-bibliotek, mens uten petsc kan jeg gå rett på mpi (hvis aktivert). dette er forklaring på at vi alltid har en MPI_Comm med PETSc.